docs/specs/ClickOnce-Signing-Algorithm.md

```diff
@@ -0,0 +1,134 @@
+1. Use [`Manifest.ResolveFiles()`](https://learn.microsoft.com/dotnet/api/microsoft.build.tasks.deployment.manifestutilities.manifest.resolvefiles?view=msbuild-17-netcore#microsoft-build-tasks-deployment-manifestutilities-manifest-resolvefiles) to resolve paths.  This method:
+   > Locates all specified assembly and file references by searching in the same directory as the loaded manifest, or in the current directory. The location of each referenced assembly and file is required for hash computation and assembly identity resolution. Any resulting errors or warnings are reported in the OutputMessages collection.
+1. Log all messages in [`Manifest.OutputMessages`](https://learn.microsoft.com/dotnet/api/microsoft.build.tasks.deployment.manifestutilities.manifest.outputmessages?view=msbuild-17-netcore).
+1. If `Manifest.OutputMessages` contains any errors, fail signing.
+1. Obtain the full path of the application manifest file from [`DeployManifest.EntryPoint`](https://learn.microsoft.com/dotnet/api/microsoft.build.tasks.deployment.manifestutilities.deploymanifest.entrypoint?view=msbuild-17-netcore)[`.ResolvedPath`](https://learn.microsoft.com/dotnet/api/microsoft.build.tasks.deployment.manifestutilities.basereference.resolvedpath?view=msbuild-17-netcore#microsoft-build-tasks-deployment-manifestutilities-basereference-resolvedpath).
+1. If the application manifest file does not exist, log a warning and stop further processing of the deployment manifest file.
+1. Read the application manifest file using `ManifestReader.ReadManifest(...)`.
+1. Set `Manifest.ReadOnly` to `true` to ensure read-only mode.
+1. Interate through both [`AssemblyReferences`](https://learn.microsoft.com/dotnet/api/microsoft.build.tasks.deployment.manifestutilities.manifest.assemblyreferences?view=msbuild-17-netcore) and [`FileReferences`](https://learn.microsoft.com/dotnet/api/microsoft.build.tasks.deployment.manifestutilities.manifest.filereferences?view=msbuild-17-netcore), manually resolve `TargetPath` property to the base path of the application manifest file.
+   * Note:  it seems like calling `Manifest.ResolveFiles()` would resolve the full file path for every file dependency in the application manifest.  However, this fails because `ResolveFiles()` assumes dependency files do not have `.deploy` file extension, but they do.  We could temporarily remove the `.deploy` extension and then call `ResolveFiles()` but a user's glob patterns might filter out files based on the `.deploy` extension.  The safest option is to resolve file paths ourselves.
+1. Copy all files from the previous step, including the application manifest file itself, to a temporary directory.
+1. Sign files in the following order:  files alongside the application manifest, the application manifest itself, the deployment manifest, then `setup.exe`.
```

Comment:
You can sign files programmatically using https://github.com/dotnet/msbuild/blob/a9d68ab58eab3c25b6378a90be8060dd1429a6ef/src/Tasks/ManifestUtil/SecurityUtil.cs#L552